flink-table/flink-table-common/src/main/java/org/apache/flink/table/utils/DateTimeUtils.java

```diff
@@ -1299,6 +1380,39 @@ private static long ceil(long a, long b) {
         }
     }
 
+    private static long ceil(long a, int b) {
+        float q = (float) b / NANOS_PER_MILLISECOND;
+        if (q > 0) {
+            return a + 1L;
+        } else {
+            return a;
+        }
+    }
+
+    private static int ceilForHighPrecision(int b) {
```

Comment:
let's rename it to something like `ceilWithNanos` rather than high precision since it is more clear what is implied here.